diff --git src/test/cpp/naxriscv/makefile src/test/cpp/naxriscv/makefile
index ae7faf9..75832f2 100644
--- src/test/cpp/naxriscv/makefile
+++ src/test/cpp/naxriscv/makefile
@@ -72,13 +72,28 @@ endif
 
 LIBS +="-lpthread -ldl -lboost_regex -lboost_system  -lpthread  -lboost_system -lboost_regex"
 
+
+ADDCFLAGS += -CFLAGS -I$(DRIVER_ROOT)/include
+
+ifeq ($(FUZZER), surgefuzz)
+	ADDCFLAGS += -CFLAGS -DFUZZER_SURGEFUZZ
+else ifeq ($(FUZZER), rfuzz)
+	ADDCFLAGS += -CFLAGS -DFUZZER_RFUZZ
+else ifeq ($(FUZZER), difuzzrtl)
+	ADDCFLAGS += -CFLAGS -DFUZZER_DIFUZZRTL
+else ifeq ($(FUZZER), directfuzz)
+	ADDCFLAGS += -CFLAGS -DFUZZER_DIRECTFUZZ
+else ifeq ($(FUZZER), blackbox)
+	ADDCFLAGS += -CFLAGS -DFUZZER_BLACKBOX
+endif
+
 all: clean run
 
 run: compile
 	./obj_dir/VNaxRiscv
 
-verilate: ${NAXRISCV_VERILOG}
-	verilator -cc  ${NAXRISCV_VERILOG}  -CFLAGS -std=c++14 -LDFLAGS -pthread  ${ADDCFLAGS}  ${FLAGS} --gdbbt ${VERILATOR_ARGS} -Wno-UNOPTFLAT -Wno-WIDTH --x-assign unique --exe ${SRCS}
+verilate:
+	verilator -cc --assert  ${NAXRISCV_VERILOG}  -CFLAGS -std=c++14 -CFLAGS -fconcepts -LDFLAGS -pthread  ${ADDCFLAGS}  ${FLAGS} --gdbbt ${VERILATOR_ARGS} -Wno-UNOPTFLAT -Wno-WIDTH --x-assign unique --exe ${SRCS}
 
 src/nax.h: ${NAXRISCV_HEADER}
 	cp ${NAXRISCV_HEADER} src/nax.h
