diff --git src/main/scala/lsu/lsu.scala src/main/scala/lsu/lsu.scala
index 1f17835d..351ffe67 100644
--- src/main/scala/lsu/lsu.scala
+++ src/main/scala/lsu/lsu.scala
@@ -1207,7 +1207,7 @@ class LSU(implicit p: Parameters, edge: TLEdgeOut) extends BoomModule()(p)
     when (will_fire_store_commit(0) || !can_fire_store_commit(0)) {
       store_blocked_counter := 0.U
     } .elsewhen (can_fire_store_commit(0) && !will_fire_store_commit(0)) {
-      store_blocked_counter := Mux(store_blocked_counter === 15.U, store_blocked_counter + 1.U, 15.U)
+      store_blocked_counter := Mux(store_blocked_counter === 15.U, 15.U, store_blocked_counter + 1.U)
     }
     when (store_blocked_counter === 15.U) {
       block_load_wakeup := true.B
